pkg/virt-handler/vm.go

```diff
@@ -466,11 +468,19 @@ func (d *VirtualMachineController) execute(key string) error {
 }
 
 func (d *VirtualMachineController) replacePVCByHostDisk(vmi *v1.VirtualMachineInstance) error {
-	// if PVC is defined then it is replaced by HostDisk
-	// PersistenVolumeClaim is mounted into pod as directory from node filesystem
-	// it is not attached as independent block device
+	// if PVC is defined and it's not a BlockMode PVC, then it is replaced by HostDisk
+	// Filesystem PersistenVolumeClaim is mounted into pod as directory from node filesystem
 	for i := range vmi.Spec.Volumes {
 		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.PersistentVolumeClaim != nil {
+
+			isBlockVolumePVC, err := util.IsPVCBlockFromClient(d.clientset, vmi.Namespace, volumeSource.PersistentVolumeClaim.ClaimName)
```

Comment:
Please fetch the PV once, then check for the `volumeMode` on the PV (not PVC). It is the task of the storage provider to create a PV with the right `volumeMode` which matches the `PVC`, so it is ok to check on the PV.

Reply:
Edit: PVC is fine, misread the code. But still fetch it only once and pass it to both functions.

Reply:
oh, good catch, fixed (see `ReplacePVCByHostDisk` in `host-disk` package now)